research/object_detection/core/target_assigner.py

```diff
@@ -1897,3 +1899,170 @@ def assign_corner_offset_targets(
 
     return (tf.stack(corner_targets, axis=0),
             tf.stack(foreground_targets, axis=0))
+
+class DETRTargetAssigner(object):
+  """Target assigner to compute classification and regression targets."""
+
+  def __init__(self, negative_class_weight=1.0):
+    """Construct Object Detection Target Assigner.
+
+    Args:
+      matcher: an object_detection.core.Matcher used to match groundtruth to
+        predicted boxes.
+      box_coder_instance: an object_detection.core.BoxCoder used to encode
+        matching groundtruth boxes with respect to predicted boxes.
+      negative_class_weight: classification weight to be associated to negative
+        boxes (default: 1.0). The weight must be in [0., 1.].
+
+    """
+    self._similarity_calc = sim_calc.DETRSimilarity()
+    self._matcher = hungarian_matcher.HungarianBipartiteMatcher()
+    self._negative_class_weight = negative_class_weight
+
+  def batch_assign(self,
+                   pred_boxes_batch,
+                   gt_box_batch,
+                   class_predictions,
```

Comment:
nit: make the names consistent -- e.g predicted_boxes, predicted_classes, grountruth_boxes, groundtruth_classes, groundtruth_weights.